File: MichiMark/Projection/EventDetail/EventDetailProjection.swift
public struct EventDetailProjection: Equatable {
    public let eventId: EventID
    public let basicInfo: BasicInfoProjection
    public let michiInfo: MichiInfoListProjection
    public let paymentInfo: PaymentInfoProjection
    public let overview: OverviewProjection
}

extension EventDetailProjection {
    static func empty(eventID: EventID) -> Self {
        .init(
            eventId: eventID,
            basicInfo: .empty(eventID: eventID),
            michiInfo: .empty,
            paymentInfo: .empty,
            overview: .empty
        )
    }
}

File: flutter/lib/features/event_detail/projection/event_detail_projection.dart
import 'package:equatable/equatable.dart';
import 'basic_info_projection.dart';
import 'michi_info_list_projection.dart';
import 'payment_info_projection.dart';

class EventDetailProjection extends Equatable {
  final String eventId;
  final BasicInfoProjection basicInfo;
  final MichiInfoListProjection michiInfo;
  final PaymentInfoProjection paymentInfo;

  const EventDetailProjection({
    required this.eventId,
    required this.basicInfo,
    required this.michiInfo,
    required this.paymentInfo,
  });

  static EventDetailProjection empty(String eventId) => EventDetailProjection(
        eventId: eventId,
        basicInfo: BasicInfoProjection.empty(eventId),
        michiInfo: MichiInfoListProjection.empty,
        paymentInfo: PaymentInfoProjection.empty,
      );

  @override
  List<Object?> get props => [eventId, basicInfo, michiInfo, paymentInfo];
}
